src/daft-recordbatch/src/file_info.rs

```diff
@@ -60,19 +62,43 @@ impl FileInfos {
 
     /// Concatenate two FileInfos together.
     pub fn extend(&mut self, new_infos: Self) {
```

Comment:
nit: rename this to `merge` as it's no longer just extending, but merging in the new file infos, and deduplicating in the process